### PR TITLE
chore: output semantic version breakdown

### PR DESCRIPTION
**Pull request by acazacu**
Output the breakdown of a the next release version. The following outputs have been added:
- major: The major component of the released version.
- minor: The minor component of the released version.
- patch: The patch component of the released version.
- pre-release: The pre-release component of the released version.
- Build: The build component of the released version.

Please refer to the [semver documentation](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string) for additional details regarding the semver breakdown.

### CODE REVIEW

File: .github/workflows/continuous-delivery.yaml [hunk 1]
```diff
@@ -39,7 +39,13 @@ jobs:
           commit-assets: |
             ./dist
           node-module: true
+      - env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN_DOTTBOTT }}
+        id: tag
+        if: steps.released.outputs.released == 'true'
+        name: Tag
+        uses: Actions-R-Us/actions-tagger@v2
```
Comment (merlinnot) [inline .github/workflows/continuous-delivery.yaml:47]: Why can't we just use git for it? Looks like we could just use `persist-credentials: true`?

Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml:47]: As in tag with the git CLI?

Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml:47]: This is the semver regex that splits a version number into relevant capture groups:

`^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`

Groups 1-3 are the major, minor and patch versions.
Group 4 is the prerelease tag.
Group 5 is the build metadata.

Our tag should look like this: `v{major}-{prerelease tag}`. How do I go about generating this string using sed? Can you lend a hand, I'm sort of drowning into the regex black hole here? 😬

Comment (merlinnot) [reply .github/workflows/continuous-delivery.yaml]: Ugh. Maybe we should have outputs with specific information? Is this exposed by Semantic?

File: .github/workflows/continuous-delivery.yaml [hunk 2]
```diff
@@ -43,3 +43,8 @@ jobs:
         if: steps.release.outputs.released == 'true'
         run: |
           echo released version: ${{ steps.release.outputs.version }}, type: ${{ steps.release.outputs.level }}
+          echo major: ${{ steps.release.outputs.major }}
+          echo minor: ${{ steps.release.outputs.minor }}
+          echo patch: ${{ steps.release.outputs.patch }}
+          echo pre-release: ${{ steps.release.outputs.pre-release }}
+          echo build: ${{ steps.release.outputs.build }}
```
Comment (merlinnot) [inline .github/workflows/continuous-delivery.yaml:50]: This should all be removed, right? I mean the entire step, why is it here?

File: action.yml [hunk 3]
```diff
@@ -27,12 +27,23 @@ inputs:
...
-  version:
-    description: Released version in format of X.Y.Z (major.minor.patch).
+  major:
+    description: Major component of the released version.
+  build:
+    description: Build component of the released version.
+  minor:
+    description: Minor component of the released version.
+  patch:
+    description: Patch component of the released version.
```
Comment (merlinnot) [inline action.yml:37]: This is neither lexicographical nor semantic release order. Can we just run a YAML sorter on this file?

File: action.yml [hunk 4]
```diff
@@ -27,12 +27,23 @@ inputs:
     description: Configures the branches that Semantic Release will release to.
     required: false
 outputs:
-  version:
-    description: Released version in format of X.Y.Z (major.minor.patch).
+  major:
+    description: Major component of the released version.
+  build:
```
Comment (merlinnot) [inline action.yml:32]: This is called `meta` in `version` field description. We should settle on one.

File: src/utilities/outputParsers.ts [hunk 5]
```diff
@@ -2,11 +2,25 @@ import { setOutput } from '@actions/core';
...
+
+interface SemVerGroups {
+  build?: string;
+  major: string;
+  minor: string;
+  patch: string;
+  preRelease?: string;
 }
 
+// eslint-disable-next-line max-statements
```
Comment (merlinnot) [inline src/utilities/outputParsers.ts:23]: Let's split it into smaller functions instead.

**Pull request merged**